code/modules/atmospherics/machinery/components/fusion/hfr_procs.dm

```diff
@@ -162,6 +162,71 @@
 		corners = list()
 	QDEL_NULL(soundloop)
 
+/obj/machinery/atmospherics/components/unary/hypertorus/core/proc/assert_gases()
+	//Assert the gases that will be used/created during the process
+
+	internal_fusion.assert_gas(/datum/gas/antinoblium)
+
+	for(var/gas_id in GLOB.meta_gas_info)
+		moderator_internal.assert_gas(gas_id)
+
+	if (!selected_fuel)
+		return
+
+	for(var/gas_id in selected_fuel.requirements) //These are the fuels
+		internal_fusion.assert_gas(gas_id)
+
+	for(var/gas_id in selected_fuel.primary_products) //These are the gases that can be produced inside the internal_fusion mix
+		internal_fusion.assert_gas(gas_id)
+
```

Comment:
Why not use `assert_gases()`?

Reply:
Good idea. I stopped after swapping out `|=` but you're right, there's still low hanging fruit to be had here

Reply:
```dm
internal_fusion.assert_gases(arglist(selected_fuel.requirements | selected_fuel.primary_products))
```
Possibly a bit verbose but I think it's still an improvement

Reply:
needing an arglist was a mistake, but that will work just fine